lib/fluent/plugin/out_amqp.rb

```diff
@@ -110,6 +111,9 @@ def configure(conf)
       unless @key || @tag_key
         raise Fluent::ConfigError, "Either 'key' or 'tag_key' must be set."
       end
+      unless @chunk_key_tag
+        raise Fluent::ConfigError, "'tag' in chunk_keys is required."
```

Comment:
This line ensure `tag` existence in chunk_keys in buffer directive:

```
<buffer tag>
  @type memory
</buffer>
```

is correct, but

```
<buffer mykey>
  @type memory
</buffer>
```

or

```
<buffer time>
  @type memory
  timekey      1h
  timekey_wait 5m 
</buffer>
```

are treated as "incorrect" with this configure.

In more detail, please refer to the document: http://docs.fluentd.org/v0.14/articles/buffer-section